src/core/src/CoreMain.py

```diff
@@ -84,19 +91,22 @@ def __init__(self, argv):
             if telemetry_writer is not None:
                 telemetry_writer.write_event("EXCEPTION: " + repr(error), Constants.TelemetryEventLevel.Error)
             if status_handler is not None:
-                composite_logger.log_debug(' - Status handler pending writes flags [I=' + str(patch_installation_successful) + ', A=' + str(patch_assessment_successful) + ']')
-                if patch_operation_requested == Constants.INSTALLATION.lower() and not patch_installation_successful:
-                    status_handler.set_installation_substatus_json(status=Constants.STATUS_ERROR)
-                    composite_logger.log_debug('  -- Persisted failed installation substatus.')
-                if not patch_assessment_successful:
-                    status_handler.set_assessment_substatus_json(status=Constants.STATUS_ERROR)
-                    composite_logger.log_debug('  -- Persisted failed assessment substatus.')
+                composite_logger.log_debug(' - Status handler pending writes flags [I=' + str(overall_patch_installation_operation_successful) + ', A=' + str(patch_assessment_successful) + ']')
 
+                # Current operation is set to either assessment or installation when these operations begin.
+                # If None is set at this point that is an indication that something went wrong before the first assessment operation could start.
+                # Logging such errors under assessment substatus since, error details within installation will point to assessment errors
+                if status_handler.get_current_operation() is None:
+                    status_handler.set_current_operation(Constants.ASSESSMENT)
```

Comment:
Good change..

Reply:
Actually, why can't this be set straight up near the beginning? The current operation (at first) will always be assessment. So it can be set right after status handler init (explicitly though, vs. default status handler initing to that). 
Is there a reason this need to happen here vs. that?